mkosi/initrd.py

```diff
@@ -169,6 +169,20 @@ def main() -> None:
 
         cmdline += ["--sandbox-tree", sandbox_tree]
 
+        # Generate crypttab with all the x-initrd.attach entries
+        if Path("/etc/crypttab").exists():
+            crypttab = []
+            for line in Path("/etc/crypttab").read_text().splitlines():
+                entry = line.split()
+                if len(entry) < 4 or entry[0].startswith("#") or "x-initrd.attach" not in entry[3]:
+                    continue
+                crypttab.append(line)
```

Comment:
Just for completeness, this would be an option to do it in a comprehension
```suggestion
            crypttab = [
                line
                for line in Path("/etc/crypttab").read_text().splitlines()
                if len(entry := line.split()) >= 4 and not entry[0].startswith("#") and "x-initrd.attach" in entry[3]
            ]
```
I think it's a bit clearer, but that's probably in the eye of the beholder because of the walrus stretches this a bit.

Reply:
ok, changed